pkg/asset/cluster/aws/aws.go

```diff
@@ -155,6 +159,67 @@ func tagSharedIAMRoles(ctx context.Context, clusterID string, installConfig *ins
 	return nil
 }
 
+// tagSharedIAMProfiles tags users BYO instance profiles so they are not destroyed by the Installer.
+func tagSharedIAMProfiles(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig) error {
+	iamProfileNames := sets.New[string]()
+
+	{
+		mpool := awstypes.MachinePool{}
+		mpool.Set(installConfig.Config.AWS.DefaultMachinePlatform)
+
+		if mp := installConfig.Config.ControlPlane; mp != nil {
+			mpool.Set(mp.Platform.AWS)
+		}
+		if len(mpool.IAMProfile) > 0 {
+			iamProfileNames.Insert(mpool.IAMProfile)
+		}
+	}
+
+	for _, compute := range installConfig.Config.Compute {
+		mpool := awstypes.MachinePool{}
+		mpool.Set(installConfig.Config.AWS.DefaultMachinePlatform)
```

Comment:
@r4f4 This is a question to understand the code base better and not a comment.

What are we accomplishing at this step by setting mpool to `installConfig.Config.AWS.DefaultMachinePlatform` within this Compute block?

Reply:
@sadasu sure, let me see if I can explain it.

So if you look at the [pkg.types.aws.MachinePool.Set](https://github.com/openshift/installer/blob/master/pkg/types/aws/machinepool.go#L47) function, it works like this:
* Let "target" be the lefthand side machine pool (in our case it's an empty machine pool `mpool := awstypes.MachinePool{}`
* For each field of `required` (in our case it's `installConfig.Config.AWS.DefaultMachinePlatform`), if that field is set, then it writes the value to our "target" machine pool.
* So at the end of `mpool.Set(installConfig.Config.AWS.DefaultMachinePlatform)`, our target `mpool`, which was empty, now has all the fields that were set in the defaultMachinePlatform machine pool.
* The same process repeats in the next line for `mpool.Set(compute.Platform.AWS)` but this time our target `mpool` has all the fields that were set for the compute machine pool in the install config + the fields that were defined for the default machine pool.

What that does, in other words, is machine pool "inheritance": get all the fields that were set for `compute.platform.aws`. For those fields that were not set, get the values from the `platform.aws.defaultMachinePlatform` machine pool instead.

In the case of this PR, the `iamProfile` field can be defined in the `compute.platform.aws.iamProfile` machine pool directly, _or_ it can be set in `platform.aws.defaultMachinePool.iamProfile`. Both controlPlane and compute "inherit" the values from the defaultMachinePlatform when their fields are not set.

Does that make sense?

Reply:
Another way of accomplishing the same thing would be to do:
```golang
for _, compute := range installConfig.Config.Compute {
    iamProfile := ""
    if installConfig.Config.AWS.DefaultMachinePlatform != nil && installConfig.Config.AWS.DefaultMachinePlatform.IAMProfile != "" {
        iamProfile = installConfig.Config.AWS.DefaultMachinePlatform.IAMProfile
    }
    if compute.Platform.AWS != nil && compute.Platform.AWS.IAMProfile != "" {
        iamProfile = compute.Platform.AWS.IAMProfile
    }

    if len(iamProfile) != "" {
        iamProfileNames.Insert(iamProfile)
    }
}
```
If find that a lot more cumbersome than just using the existing machine pool "inheritance" feature.

Reply:
@r4f4 thanks for the explanation. 
I was expecting machine pool "inheritence" to work in a way that if compute.platform.aws is present, then all values are taken from there. Same with control plane. And we would default to `DefaultMachinePlatform` only for the one or both of control and compute machine pools that are absent.

Reply:
@sadasu I see. I think the idea is more like: use these default fields unless they are overridden in the controlPlane/compute machine pools.